src/gt4py/next/ffront/decorator.py

```diff
@@ -111,33 +100,14 @@ def definition(self):
 
     @functools.cached_property
     def past_stage(self):
-        if self.backend is not None and self.backend.transformer is not None:
-            return self.backend.transformer.func_to_past(self.definition_stage)
-        return next_backend.DEFAULT_TRANSFORMS.func_to_past(self.definition_stage)
+        if self.backend is not None and self.backend.transforms_prog is not None:
+            return self.backend.transforms_prog.func_to_past(self.definition_stage)
+        return next_backend.DEFAULT_PROG_TRANSFORMS.func_to_past(self.definition_stage)
 
     def __post_init__(self):
-        function_closure_vars = transform_utils._filter_closure_vars_by_type(
-            self.past_stage.closure_vars, GTCallable
-        )
-        misnamed_functions = [
-            f"{name} vs. {func.id}"
-            for name, func in function_closure_vars.items()
-            if name != func.__gt_itir__().id
-        ]
-        if misnamed_functions:
-            raise RuntimeError(
-                f"The following symbols resolve to a function with a mismatching name: {','.join(misnamed_functions)}."
-            )
-
-        undefined_symbols = [
-            symbol.id
-            for symbol in self.past_stage.past_node.closure_vars
-            if symbol.id not in self.past_stage.closure_vars
-        ]
-        if undefined_symbols:
-            raise RuntimeError(
-                f"The following closure variables are undefined: {', '.join(undefined_symbols)}."
-            )
+        if self.backend is not None and self.backend.transforms_prog is not None:
+            self.backend.transforms_prog.past_lint(self.past_stage)
+        return next_backend.DEFAULT_PROG_TRANSFORMS.past_lint(self.past_stage)
```

Comment:
Is the answer to the question above this line? I.e. when linting, but the backend doesn't have transforms_prog and therefore no linting, then we fall back to the linting via `DEFAULT_PROG_TRANSFORMS` and therefore need a `past_stage`?

Reply:
Exactly. This is to be backwards compatible. Linting even if there is no backend. Also this ensures it doesn't break when the dummy `no_backend` is used in testing.

Reply:
I am not sure where to document that (it occurs in so many places).

Reply:
I put a comment on the first occurrence. Is that enough?